AppWidget/README.md

```diff
@@ -1,396 +1,81 @@
+* Android SDK 31
+* Android Studio Chipmunk or newer
 
-* [`targetCellWidth`](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo#targetCellWidth) and [`targetCellHeight`](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo#targetCellHeight): Defines the target size of the widget in terms of launcher grid cells. If defined, these attributes are used instead of minWidth or minHeight.
-* [`maxResizeWidth`](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo#maxResizeWidth) and [`maxResizeHeight`](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo#maxResizeHeight): Defines the maximum size the launcher allows the user to resize the widget.
+# Getting Started
 
-The following XML describes how to use the sizing attributes.
+This sample uses the Gradle build system. To build this project, use the
+"gradlew build" command or use "Import Project" in Android Studio.
 
-[`xml/app_widget_info_checkbox_list.xml`](app/src/main/res/xml/app_widget_info_checkbox_list.xml\#L38)
+The sample contains 3 type of widgets:
 
-```xml
-<appwidget-provider
-  android:maxResizeWidth="240dp"
-  android:maxResizeHeight="220dp"
-  android:minWidth="180dp"
-  android:minHeight="180dp"
-  android:minResizeWidth="180dp"
-  android:minResizeHeight="180dp"
-  android:targetCellWidth="3"
-  android:targetCellHeight="3"
-  ... />
-```
+- ToDo list: showcasing how to create a list of items
+- Buttons: showcasing how to use the new CompoundedButtons and handle click events
+- Weather: showcasing a "real-life" weather widget.
 
-<img src="screenshots/widget_resizing.png"
-      alt="Screenshot of a widget being resized"
-      title="Screenshot of a widget being resized" />
+Those are implemented in two packages:
 
-### Provide responsive layouts [(docs)](https://developer.android.com/about/versions/12/features/widgets#provide-responsive-layouts)
+- [rv](app/src/main/java/com/example/android/appwidget/rv) for RemoteViews implementation
+- [glance](app/src/main/java/com/example/android/appwidget/glance) for Glance implementation
 
-If the layout needs to change depending on the size of the widget, we recommend creating a small set of layouts, each valid for a range of sizes. (If this isn’t possible, another option is to provide layouts based on the exact widget size at runtime.)
+In addition, the [MainActivity](app/src/main/java/com/example/android/appwidget/MainActivity.kt)
```

Comment:
Suggest:

In addition to building and running these samples, you can use our `appwidget` [`MainActivity`](app/src/main/java/com/example/android/appwidget/MainActivity.kt) to understand how to submit a request to the launcher to pin the widget.